rclcpp_examples/test/test_executable_linking.py.in

```diff
@@ -0,0 +1,27 @@
+# generated from rclcpp_examples/test/test_executable_linking.py.in
+
+import os
+import shlex
+import subprocess 
+
+def setup():
+    os.environ['OSPL_VERBOSITY'] = '8'  # 8 = OS_NONE
+
+
+def test_executable():
+
+    rmw_implementation = '@rmw_implementation@'
+    executables = '@RCLCPP_EXAMPLES_EXECUTABLE@'.split(';')
+    for exe in executables:
+        # Check that the executable is linked against the claimed rmw implementation
+        proc1 = subprocess.Popen(shlex.split('ldd %s'%exe), stdout=subprocess.PIPE)
+        proc2 = subprocess.Popen(shlex.split('grep -c %s'%rmw_implementation), stdin=proc1.stdout,
+                                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        proc1.stdout.close()
+        out, err = proc2.communicate()
+        links = int(out.decode("ASCII").strip())
+        assert links > 0, 'Executable is not linked against %s RMW implementation. %i links found.' \
+            % (rmw_implementation, links)
```

Comment:
Since you never know how the placeholders are substituted the resulting text might be difficult to read / parse by a human. E.g. `rmw_implementation` having the value "any" (weird example I know :wink:).

Therefore you could wrap the `%s` placeholder in single quote to clarify the semantic (then the whole string needs double quotes to avoid escaping):

    assert links > 0, "Executable is not linked against RMW implementation '%s'. %i links found." \
        % (rmw_implementation, links)